diff --git a/node_modules/prosemirror-tables/dist/index.cjs b/node_modules/prosemirror-tables/dist/index.cjs
--- a/node_modules/prosemirror-tables/dist/index.cjs
+++ b/node_modules/prosemirror-tables/dist/index.cjs
@@ -2104,7 +2104,7 @@ function insertCells(state, dispatch, tableStart, rect, cells) {
 		tr.replace(tr.mapping.slice(mapFrom).map(from + tableStart), tr.mapping.slice(mapFrom).map(to + tableStart), new prosemirror_model.Slice(cells.rows[row - top], 0, 0));
 	}
 	recomp();
-	tr.setSelection(new CellSelection(tr.doc.resolve(tableStart + map.positionAt(top, left, table)), tr.doc.resolve(tableStart + map.positionAt(bottom - 1, right - 1, table))));
+	tr.setSelection(new CellSelection(tr.doc.resolve(tableStart + map.map[top * map.width + left]), tr.doc.resolve(tableStart + map.map[(bottom - 1) * map.width + right - 1])));
 	dispatch(tr);
 }
 
diff --git a/node_modules/prosemirror-tables/dist/index.js b/node_modules/prosemirror-tables/dist/index.js
--- a/node_modules/prosemirror-tables/dist/index.js
+++ b/node_modules/prosemirror-tables/dist/index.js
@@ -2104,7 +2104,7 @@ function insertCells(state, dispatch, tableStart, rect, cells) {
 		tr.replace(tr.mapping.slice(mapFrom).map(from + tableStart), tr.mapping.slice(mapFrom).map(to + tableStart), new Slice(cells.rows[row - top], 0, 0));
 	}
 	recomp();
-	tr.setSelection(new CellSelection(tr.doc.resolve(tableStart + map.positionAt(top, left, table)), tr.doc.resolve(tableStart + map.positionAt(bottom - 1, right - 1, table))));
+	tr.setSelection(new CellSelection(tr.doc.resolve(tableStart + map.map[top * map.width + left]), tr.doc.resolve(tableStart + map.map[(bottom - 1) * map.width + right - 1])));
 	dispatch(tr);
 }
 
@@ -2623,4 +2623,3 @@ function tableEditing({ allowTableNodeSelection = false } = {}) {
 
 //#endregion
 export { CellBookmark, CellSelection, ResizeState, TableMap, TableView, clipCells as __clipCells, insertCells as __insertCells, pastedCells as __pastedCells, addColSpan, addColumn, addColumnAfter, addColumnBefore, addRow, addRowAfter, addRowBefore, cellAround, cellNear, colCount, columnIsHeader, columnResizing, columnResizingPluginKey, deleteCellSelection, deleteColumn, deleteRow, deleteTable, findCell, findCellPos, findCellRange, findTable, fixTables, fixTablesKey, goToNextCell, handlePaste, inSameTable, isInTable, mergeCells, moveCellForward, moveTableColumn, moveTableRow, nextCell, pointsAtCell, removeColSpan, removeColumn, removeRow, rowIsHeader, selectedRect, selectionCell, setCellAttr, splitCell, splitCellWithType, tableEditing, tableEditingKey, tableNodeTypes, tableNodes, toggleHeader, toggleHeaderCell, toggleHeaderColumn, toggleHeaderRow, updateColumnsOnResize };
-//# sourceMappingURL=index.js.map
\ No newline at end of file
